python/ribasim/ribasim/model.py

```diff
@@ -170,9 +181,18 @@ def _save(self, directory: DirectoryPath, input_dir: DirectoryPath):
         for sub in self._nodes():
             sub._save(directory, input_dir)
 
+    def set_crs(self, crs: str) -> None:
+        self.edge.df = self.edge.df.set_crs(crs)
+        for sub in self._nodes():
+            if sub.node.df is not None:
+                sub.node.df = sub.node.df.set_crs(crs)
+            for table in sub._tables():
+                if isinstance(table, SpatialTableModel) and table.df is not None:
+                    table.df = table.df.set_crs(crs)
+
     def node_table(self) -> NodeTable:
         """Compute the full NodeTable from all node types."""
-        df_chunks = [node.node.df for node in self._nodes()]
+        df_chunks = [node.node.df.set_crs(self.crs) for node in self._nodes()]
```

Comment:
Together with `_save` and `set_crs` this seems overkill? Essentially, when getting the node_table (used in _save), the crs is always set to self.crs, which follows from the toml. `_save` only needs to set the edge crs, and the `set_crs` method can go?

Reply:
You are correct, the latest iteration doesn't need that.

Reply:
Had to re-introduce after all, other functions depend on it.